### PR TITLE
Tp 794 install google tag manager in to service

### PR DESCRIPTION
**Pull request by gabelton**
## Why

This change is required, in order to support go-live for the service

## What

- In _generic.jinja conditionally add Google Tag Manager script into <head> and noscript into <body> on the basis of whether GOOGLE_ANALYTICS_ID environment variable has been set.
- In settings/common.py add GOOGLE_ANALYTICS_ID environment variable. This should be set in AWS EBS for production only.

## Checklist
- Needs GOOGLE_ANALYTICS_ID to be set as environment variable in AWS EBS prod before deployment.

(Original ticket https://uktrade.atlassian.net/browse/TP-794)

### CODE REVIEW

File: common/jinja2/layouts/_generic.jinja [hunk 1]
```diff
@@ -1,6 +1,18 @@
...
 
 {% block head %}
+  {% if settings.GOOGLE_ANALYTICS_ID %}
+    <!-- Google Tag Manager -->
+      <script>
+        (function(w,d,s,l,i){w[l]=w[l]||[];w[l].push({'gtm.start':
+        new Date().getTime(),event:'gtm.js'});var f=d.getElementsByTagName(s)[0],
+        j=d.createElement(s),dl=l!='dataLayer'?'&l='+l:'';j.async=true;j.src=
+        'https://www.googletagmanager.com/gtm.js?id='+i+dl;f.parentNode.insertBefore(j,f);
+        })(window,document,'script','dataLayer', {{settings.GOOGLE_ANALYTICS_ID}});
```
Comment (simonwo) [inline common/jinja2/layouts/_generic.jinja:11]: Does the `{{ … }}` need to have string quotes around it? Or does Jinja know to do that automatically?

Comment (gabelton) [reply common/jinja2/layouts/_generic.jinja:11]: I think Jinja does it automatically. After adding this I had a play around in my local and I was able to pass through a value.

**Comment by stuaxo** [comment]
Make sure to add any new environment variables note in the environment variables section in the README.md.

File: settings/common.py [hunk 2]
```diff
@@ -346,6 +346,9 @@
     },
 }
 
+# -- Google Tag Manager
```
Comment (stuaxo) [inline settings/common.py:349]: Make sure to document this in the README as well.

If we have a page on our RTFM docs, it's worth making a link to it, from here or from the README.

**Review by stuaxo — APPROVED** [review]

File: common/jinja2/layouts/_generic.jinja (hunk 1 above)
Comment (stuaxo) [reply common/jinja2/layouts/_generic.jinja:11]: To make the indentation consistent, I think it should be `{{ settings.GOOGLE_ANALYTICS_ID }}`

Comment (andyhd) [reply common/jinja2/layouts/_generic.jinja:11]: I think it should have quotes. I'm pretty sure Jinja does not do it automatically.
```python
>>> jinja2.Template("""<script>var some_var = {{value}};</script>""").render({"value": "a_string"})
'<script>var some_var = a_string;</script>'
```

File: settings/common.py [hunk 3]
```diff
@@ -346,6 +346,9 @@
...
+GOOGLE_ANALYTICS_ID = os.environ.get("GOOGLE_ANALYTICS_ID", None)
```
Comment (andyhd) [inline settings/common.py:350]: `None` is the default default value, so you can just do `.get("GOOGLE_ANALYTICS_ID")`

Obviously doesn't need fixing, I just wanted to write "default default" 😄

**Review by andyhd — CHANGES_REQUESTED** [review]
I think @simonwo was right

File: common/jinja2/layouts/_generic.jinja (hunk 1 above)
Comment (andyhd) [reply common/jinja2/layouts/_generic.jinja:11]: I just thought - maybe check whether you have quotes in the `settings.GOOGLE_ANALYTICS_ID` string value (eg: your env var is `GOOGLE_ANALYTICS_ID="'some-id-string'"`)

File: common/jinja2/layouts/_generic.jinja [hunk 4]
```diff
@@ -31,7 +31,7 @@
   {% if settings.GOOGLE_ANALYTICS_ID %}
     <!-- Google Tag Manager (noscript) -->
       <noscript>
-        <iframe src="https://www.googletagmanager.com/ns.html?id={{settings.GOOGLE_ANALYTICS_ID}}"height="0" width="0" style="display:none;visibility:hidden"></iframe>
+        <iframe src="https://www.googletagmanager.com/ns.html?id="{{ settings.GOOGLE_ANALYTICS_ID }}""height="0" width="0" style="display:none;visibility:hidden"></iframe>
```
Comment (simonwo) [inline common/jinja2/layouts/_generic.jinja:34]: Nah, not this one… the quotes will end up ending the `href` element. It should have just been as it was in the previous version.

File: common/jinja2/layouts/_generic.jinja (hunk 1 above)
Comment (gabelton) [reply common/jinja2/layouts/_generic.jinja]: The env var in Vault doesn't have quotes around it

**Review by andyhd — APPROVED** [review]

**Pull request merged**